src/modules/lifecycle_management.json

```diff
@@ -0,0 +1,254 @@
+                    }
+                },
+                {
+                    "name": "state",
+                    "schema": {
+                        "$ref": "#/components/schemas/State"
+                    }
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "const": null
+                }
+            },
+            "examples": [
+                {
+                    "name": "Set application lifecycle state to inactive",
+                    "params": [
+                        {
+                            "name": "state",
+                            "value": "inactive"
+                        },
+                        {
+                            "name": "appId",
+                            "value": "SomeAppId"
+                        }
+                    ],
+                    "result": {
+                        "name": "defaultResult",
+                        "value": null
+                    }
+                }
+            ]
+        },
+        {
+            "name": "onRequestReady",
+            "summary": "Tells the provider that an app has said they are ready",
+            "params": [
+                {
+                    "name": "listen",
+                    "summary": "Whether or not to listen for requests to this interface method.",
+                    "schema": {
+                        "type": "boolean"
+                    }
+                }
+            ],
+            "tags": [
+                {
+                  "name": "capabilities",
+                  "x-provides": "xrn:firebolt:capability:app:lifecycle",
+                  "x-allow-focus": false
+                }
+              ],
+            "result": {
+                "name": "ready",
+                "summary": "The app has said they are ready",
+                "schema": {
+                    "oneOf": [
+                        {
+                          "$ref": "https://meta.comcast.com/firebolt/types#/definitions/ListenResponse"
+                        },
+                        {
+                          "$ref": "#/components/schemas/LifecycleResult"
+                        }
+                    ]            
+                }
+            },
+            "examples": [
+                {
+                    "name": "Default Example",
+                    "params": [
+                        {
+                            "name": "listen",
+                            "value": true
+                        }
+                    ],
+                    "result": {
+                        "name": "Default Result",
+                        "value": {
+                          "parameters": {
+                            "appId": "SomeAppId"
+                          }
+                        }
+                    }
+                }
+            ]
+        },
+        {
+            "name": "onRequestClose",
```

Comment:
Lifecycle.close() has a required parameter called `reason` with three values. I think we need to forward that reason to the LifecycleManagement provider.

Reply:
yeah good point @jlacivita